reworking template docs (#606)
actualbudget/docs: docs/experimental/goal-templates.md

        #template 150 up to 500
        #template-1 300

### Notes on all templates

- Don't add a currency symbol to your template amounts.
- Number formats that use comma for the decimal separator are not supported (e.g., 123,45). You must use 123.45.
- Thousands separators are not supported (e.g., 1,234). You must use 1234.

> - By default templates do not consider available funds when being applied. Use template priorities to not budget more than is available.

Reviewer: ```suggestion
- By default, templates do not consider available funds when being applied. Use template priorities to ensure only the amount available to budget is assigned.
```

| Vacation Fund   | #template remainder   | 16.66         |
| Investment Fund | #template remainder 2 | 33.34         |

### Goal Directive

> This option is unique enough to warrant its own directive `#goal` instead of the standard `#template` option.

Reviewer: ```suggestion
This option is unique enough to warrant its own directive, `#goal`, instead of the standard `#template` option.
```
